Specs/Scene/Cesium3DTilesetSpec.js

```diff
@@ -2901,6 +2901,53 @@ describe(
       );
     });
 
+    it("handle else case when applying conditional color style to a tileset", function () {
+      return Cesium3DTilesTester.loadTileset(scene, withBatchTableUrl).then(
+        function (tileset) {
+          tileset.style = new Cesium3DTileStyle({
+            color: {
+              conditions: [["${id} > 0", 'color("black")']],
+            },
+          });
+          expect(
+            tileset.root.content.getFeature(0).hasProperty('color("white")')
+          );
+          expect(
+            tileset.root.content.getFeature(1).hasProperty('color("black")')
+          );
```

Comment:
These two checks are actually no-ops. `expect(.......).toBe(true)` needs to be added. After making that change you'll see that the tests fail.

`hasProperty` is meant for querying properties that are stored in a content's [batch table](https://github.com/CesiumGS/3d-tiles/tree/master/specification/TileFormats/BatchTable). You can see the full list of properties when you click a feature in CesiumJS, e.g. click a building in [3D Tiles Adjust Height](https://sandcastle.cesium.com/?src=3D%20Tiles%20Adjust%20Height.html&label=3D%20Tiles) and it will show `id`, `Longitude`, `Latitude`, and `Height`.

`getFeature(0).hasProperty("id")` would return true but `getFeature(0).hasProperty('color("white")')` would not, unless for some reason the 3D Tiles had a property called `color("white")`.

To get a feature's color or show, which are actually CesiumJS specific constructs, you can use the `color` and `show` getters.

Reply:
And instead of `.toBe(true)` something like this is better: 

`expect(tileset.root.content.getFeature(0).color).toEqual(Color.WHITE);`

Reply:
@lilleyse I had originally written the spec using code similar to your recent suggestions: 

spec: `"handle else case when applying conditional color style to a tileset"`
`expect(tileset.root.content.getFeature(0).color).toEqual(Color.WHITE);`
`expect(tileset.root.content.getFeature(1).color).toEqual(Color.BLACK);`

and 

spec: `"handle else case when applying conditional show to a tileset"`
`expect(tileset.root.content.getFeature(0).show).toBe(true);`
`expect(tileset.root.content.getFeature(1).show).toBe(false);`

However, I noticed that both specs failed even though my sandcastle demos seemed to have the desired functionality. It seems like the `if` case conditions are not being applied. Any suggestions?

Reply:
Add `scene.renderForSpecs();` after setting the style and the test should pass. The style will be applied during the update.

Reply:
The styles in `handle else case when applying conditional show to a tileset` need to be fixed. `show` does not have a constructor in the styling language like `color` does.

See https://github.com/CesiumGS/3d-tiles/tree/master/specification/Styling#styling-features. Just `"true"` and `"false"` alone should work.

Reply:
@lilleyse I updated the specs:
`"handle else case when applying conditional color style to a tileset"`
and 
`"handle else case when applying conditional show to a tileset"`.

It seems like adding `scene.renderForSpecs();` after setting the style was the missing piece. One other thing, in `"handle else case when applying conditional show to a tileset"` we see that the else case defaults to `false` - is this what we envisioned?

Reply:
Ah that's because the new `show` tests are not using condition expressions

Reply:
@lilleyse We should be good to go now. I apologize for this last step taking so long. Thank you very much for your help!

Reply:
No problem, all part of the learning experience 🙂